src/useCases/deleteEquipment/deleteEquipmentUseCase.ts

```diff
@@ -0,0 +1,87 @@
+import { Equipment } from '../../db/entities/equipment'
+
+import { UseCase, UseCaseReponse } from '../protocol/useCase'
+
+import { EquipmentRepositoryProtocol } from '../../repository/protocol/equipmentRepositoryProtocol'
+
+export type DeleteEquipmentUseCaseData = {
+  id: string
+}
+
+export class NullFieldsError extends Error {
+  constructor() {
+    super('Um ou mais campos obrigatórios possuem valores nulos.')
+    this.name = 'NullFieldsError'
+  }
+}
+
+export class InvalidEquipmentError extends Error {
+  constructor() {
+    super('ID fornecido inválido.')
+    this.name = 'InvalidEquipmentError'
+  }
+}
+
+export class TimeLimitError extends Error {
+  constructor() {
+    super('Tempo limite para operação excedido.')
+    this.name = 'TimeLimitError'
+  }''
```

Comment:
remover aspas desnecessárias
```suggestion
  }
```